neuralprophet/utils.py

```diff
@@ -1,35 +1,43 @@
 import os
 import sys
 import math
+import importlib
 import numpy as np
 import pandas as pd
 import torch
 from collections import OrderedDict
 from neuralprophet import hdays as hdays_part2
 from neuralprophet import utils_torch
+from neuralprophet import time_net
 import holidays as pyholidays
 import warnings
 import logging
+import pytorch_lightning as pl
 
 log = logging.getLogger("NP.utils")
 
 
-def save(model, path):
+def save(forecaster, path):
     """save a fitted np model to a disk file.
 
     Parameters
     ----------
-        model : np.forecaster.NeuralProphet
-            input model that is fitted
+        forecaster : np.forecaster.NeuralProphet
+            input forecaster that is fitted
         path : str
             path and filename to be saved. filename could be any but suggested to have extension .np.
     Examples
     --------
     After you fitted a model, you may save the model to save_test_model.np
         >>> from neuralprophet import save
-        >>> save(model, "test_save_model.np")
+        >>> save(forecaster, "test_save_model.np")
     """
-    torch.save(model, path)
+    # Remove non-serializable components (model, trainer, metrics)
+    for attr in ["metrics", "model", "trainer"]:
+        setattr(forecaster, attr, None)
+    # Add the model back in after saving (workaround for PyTorch Lightning)
+    forecaster.model = time_net.TimeNet.load_from_checkpoint(forecaster.metrics_logger.checkpoint_path)
+    torch.save(forecaster, path)
```

Comment:
The Lightning objects cannot directly be serialized (because some tensors are still in grad-mode), thus they are removed from the NeuralProphet object. Workaround: After reading the model from a checkpoint however, we can make it part of the forecaster again.